src/si/helpers.stanza

```diff
@@ -88,6 +89,143 @@ public defn topo-net (pair:KeyValue<JITXObject,JITXObject|KeyValue>):
     (p2:JITXObject):
       topo-net(p1, p2)
 
+doc: \<DOC>
+A wrapper for `KeyValue` indicating that the given pair of
+objects should be connected by a short trace.
+<DOC>
+public deftype ShortTrace <: KeyValue
+public defn ShortTrace (kv:KeyValue) :
+  new ShortTrace :
+    defmethod key (this) : key(kv)
+    defmethod value (this) : value(kv) 
+defmethod print (o:OutputStream, st:ShortTrace) :
+  print(o, "ShortTrace(%_ => %_)" % [key(st) value(st)])
+
+doc: \<DOC>
+  This function will construct chained topology net consisting of
+  a sequence of "bus segments".
+
+  Given a sequence of key-value pairs of "bus segments", this function
+  will connect them, generate topology segments, and constrain the trace
+  length as necessary.
+
+  A "bus segment" is either:
```

Comment:
`segment` here implies a start, end and edge of the topology. I think we might want to consider using `node` here instead of `segment`.